### PR TITLE
Ari - scissors✂️ 

### PR DESCRIPTION
**Pull request by arigo7**
First attempt

### CODE REVIEW

File: swap_meet/item.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        
+    def condition_description(self): 
+        """
+        Describes the condition in words based on the value, assuming 
+        they all range from 0 to 5
+        Returns:
+            String: Representing the default condition based on each
+            class category 
+        """        
+        condition_cat = [(0, "You probably want a glove "), (1, "Heavily used and loved"), \
```
Comment (jmaddox19) [inline swap_meet/item.py:23]: Love these :D

File: swap_meet/item.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        ( 2, "Check it out!"), (3, "Giving it its' all"), (4, "Living its' best life"), (5,  "Minty!")]
+        message = "Item's Condition: + "
+        if self.condition == None or self.condition < 0 or self.condition > 5:
```
Comment (jmaddox19) [inline swap_meet/item.py:26]: Love that you handle this edge-case just in case!

File: swap_meet/item.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            return message + "It's a mixed bag, sorry can't tell you exactly!"
+        for i in range(len(condition_cat)):
```
Comment (jmaddox19) [inline swap_meet/item.py:28]: This is totally great! I just want to point out that it could be made even slightly more efficient by making `condition_cat` a dict, with the condition numbers as the keys.

File: swap_meet/vendor.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+    def get_by_category(self, category):
+        """
+        Returns a list of Items in the inventory with that category  
+        Args:
+            category (string): attribute of class Item
+        Returns:
+            list: list of Items in the inventory with the category
+        """        
+        return [item for item in self.inventory if item.category == category]
```
Comment (jmaddox19) [inline swap_meet/vendor.py:41]: Yay list comprehensions!

File: swap_meet/vendor.py [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+            their_item (Item): an instance of friend_vendor (Vendor) 
+            Item 
+        Returns:
+            Boolean: True if swap is successful, False otherwise.
+        """        
+        friend_list = friend_vendor.inventory  
+        my_list = self.inventory  
+        if their_item in friend_list and my_item in my_list \
+            and their_item != None and my_item != None and my_list != []\
+            and friend_list !=[]:
```
Comment (jmaddox19) [inline swap_meet/vendor.py:61]: I love how thorough this conditional is! This covers edge cases the tests don't by making sure the items themselves aren't `None`.
That said, I want to point out that `my_list != [] and friend_list !=[]` are both not really necessary because the checks for `their_item in friend_list and my_item in my_list` will return false if either of the lists are empty

File: swap_meet/vendor.py [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+        friend_list = friend_vendor.inventory  
+        my_list = self.inventory 
+        if  my_list != [] and friend_list !=[]:
+            their_removed = friend_vendor.remove(friend_list[0])
+            my_removed  = self.remove(self.inventory[0]) 
+            friend_list.append(my_removed)
+            my_list.append(their_removed)
+            return True
+        else:
+            return False
```
Comment (jmaddox19) [inline swap_meet/vendor.py:88]: This code looks very similar to `swap_items`! Seems like there may be an opportunity to DRY up the code by calling `swap_items` from this method!

File: swap_meet/vendor.py [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+            other (Vendor): instance of Vendor to trade with
+            my_priority (string): category that this Vendor wants to receive
+            their_priority (string): category that other Vendor wants to
+            receive
+        Returns:
+            Boolean: True if swap is successful, False otherwise.
+        """        
+        my_item = self.get_best_by_category(their_priority)  
+        their_item = other.get_best_by_category(my_priority) 
+        swap = self.swap_items(other, my_item, their_item)
```
Comment (jmaddox19) [inline swap_meet/vendor.py:122]: Glad that you made use of existing methods here!

File: tests/test_wave_01.py [hunk 8]
```diff
@@ -2,52 +2,65 @@
 # The following line imports the Vendor class from the module vendor inside the swap_meet package.
 from swap_meet.vendor import Vendor
 
-
-def test_vendor_has_inventory():
+# TEST 1 = PASSED
+# @pytest.mark.skip(reason="no way of currently testing this")
+def test_vendor_has_inventory(): # has
+    # arrange, act -
```
Comment (jmaddox19) [inline tests/test_wave_01.py:8]: This isn't a big deal but in the future, it'd be preferred that changes to the tests don't get pushed. You can make sure they never get pushed by never using `git add` on them in the first place. In other words by specifying specific files instead of using `git add .`

**Review by jmaddox19 — COMMENTED** [review]
Looks great! Just a few little comments and things to be aware of for the future below.
